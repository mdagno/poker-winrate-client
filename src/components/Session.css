.detailedView {
  margin-left: auto;
  margin-right: auto;
  margin-top: 30px;
  color: 222;
  display: flex;
  flex-direction: column;
  width: 80%;
  padding: 10px;
  background-color: #ccc;
  border-radius: 5px;
  border: 3px outset white;
}

.detailedView h3{
  text-align: center;
}

.detailedView p {
  border: 2px inset white;
  padding: 0.3em;
}

.gameTypes {
  list-style: none;
  display: flex;
  padding: 0;
  margin: 0;
}

.gameTypes li {
  width: 50%;
  border: 2px inset white;
  text-align: center;
}

.cashMoney {
  list-style: none;
  display: flex;
  flex-direction: row;
  justify-content: center;
  margin: 0;
  padding: 0;
}

.cashMoney li {
  width: 50%;
}

p {
  text-align: center;
}

#notes {
  text-align: left;
}