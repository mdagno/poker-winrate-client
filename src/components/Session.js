import React from 'react';

export default class Session extends React.Component {
  render() {
    return(
      <p>filler</p>
    )
  }
}