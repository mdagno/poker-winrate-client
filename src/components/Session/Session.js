import React from 'react';
import ApiContext from '../../ApiContext';
import ApiService from '../../services/api-service';
import { Link } from 'react-router-dom';
import moment from 'moment';
import './Session.css';


export default class Session extends React.Component {
  static contextType = ApiContext;
  static defaultProps = {
    match: { params: {} }
  }

  componentDidMount() {
    ApiService.getSessionById(this.props.match.params.session_id)
    .then(session => this.context.setSession(session))
  }

  componentWillUnmount(){
    this.context.clearSession();
  }

  handleDeleteSession = (sessionId) => {
    ApiService.deleteSession(sessionId)
    .then(() => {
      this.context.deleteSessionContext(sessionId)
      this.props.history.push('/sessions')
    });
    
  }

  handleSubmitChanges = (e, sessionId, newSessionValues) => {
    e.preventDefault();
    ApiService.updateSession(sessionId, newSessionValues)
  }

  renderSessionView = () => {
    let sessionView = '';
    let session = this.context.currentSession;
    if (!session) {
      sessionView = <p>Loading...</p>
    }

    else {
      sessionView = (
        <div className='sessionDetails'>
        <h3>{moment(session.date_played).format('MMMM Do YYYY, h:mm:ss a')}</h3>
        <ul className='gameTypes'>
        <li>{session.game_type_one}</li>
        <li>{session.game_type_two}</li>
        </ul>
        <p>${session.small_blind} / ${session.big_blind}</p>
        <p>Time played: {session.session_length} hours</p>
        <ul className='cashMoney'>
          <li>Buy in: ${session.buy_in}</li>
          {
            (session.cashed_out - session.buy_in > 0) ? 
            <li className='winning'>+${session.cashed_out-session.buy_in}</li> : 
            <li className='losing'>-${Math.abs(session.cashed_out-session.buy_in)}</li>
          }
          <li>Cashed out: ${session.cashed_out}</li>
        </ul>
        <p id='notes'>{session.notes}</p>
        <Link to={`/sessions/${this.props.match.params.session_id}/edit`}>
        <button>Edit</button>
        </Link>
        <button onClick={() => this.handleDeleteSession(session.id)}>Delete</button>
        </div>
      )
    }
      return sessionView
    }
  
  render() {
    return(
      <div className='detailedView'>
        {this.renderSessionView()}
      </div>
    )
  }
}