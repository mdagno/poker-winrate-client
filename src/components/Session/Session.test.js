import React from 'react';
import Session from './Session'
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import ReactDOM from 'react-dom'

describe('Session component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
       <Session />
      </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Session renders correctly', () => {
    const tree = renderer
      .create(
      <BrowserRouter>
       <Session />
      </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
  
})
