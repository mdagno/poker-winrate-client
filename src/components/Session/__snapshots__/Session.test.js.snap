// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Session component Session renders correctly 1`] = `
<div
  className="detailedView"
>
  <div
    className="sessionDetails"
  >
    <h3>
      January 9th 2020, 2:56:37 am
    </h3>
    <ul
      className="gameTypes"
    >
      <li />
      <li />
    </ul>
    <p>
      $
       / $
    </p>
    <p>
      Time played: 
       hours
    </p>
    <ul
      className="cashMoney"
    >
      <li>
        Buy in: $
      </li>
      <li
        className="losing"
      >
        -$
        NaN
      </li>
      <li>
        Cashed out: $
      </li>
    </ul>
    <p
      id="notes"
    />
    <a
      href="/sessions/undefined/edit"
      onClick={[Function]}
    >
      <button>
        Edit
      </button>
    </a>
    <button
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</div>
`;
